### PR TITLE
DOCS-2221: TC Updating/organizing LDAP

### PR DESCRIPTION
**Pull request by BonnieF**
Thanks for contributing to TrueNAS documentation! By opening a Pull Request, you're acknowledging that your changes will be distributed under the [Creative Commons 4.0](https://creativecommons.org/licenses/by-nc-sa/4.0/) license.

### CODE REVIEW

**Comment by Mrt134** [comment]
@beanpole135 I added you as a reviewer for whenever Bonnie is ready. She'll remove the WIP label.

**Comment by BonnieF** [comment]
I am waiting for a review from the TrueCommand team to confirm the updated information is accurate.

File: static/includes/TrueCommand/2.0/TrueCommandLDAP.md.part [hunk 1]
```diff
@@ -15,6 +15,64 @@ Multiple LDAP servers and Domains can be added.
 
 ![Add LDAP Server](/images/TrueCommand/2.0/AdministrationConfigurationLDAPAddServer.png "Add LDAP Server")
 
+| Field | Value |
+| ----- | ------------- |
+| LDAP Server URL (string, Required) | IP or DNS name of the LDAP server, with port number on the end.
+| | Example: "ldap.mycorp.com:389" |
```
Comment (beanpole135) [inline static/includes/TrueCommand/2.0/TrueCommandLDAP.md.part:21]:
```suggestion
| | Example: "ldap.mycorp.com:636" (SSL port is typically 636 for AD/LDAP)|
```

File: static/includes/TrueCommand/2.0/TrueCommandLDAP.md.part [hunk 2]
```diff
@@ -15,6 +15,64 @@ Multiple LDAP servers and Domains can be added.
...
+
+**Direct Bind**
+In this method, the "domain" and "user_id_field" are used to create a static domain string which is then utilized to attempt to authenticate the user.
+
+Example:
+
+* { 
+* "domain" : "dc=mycorp,dc=com", 
+* "user_id_field" : "uid" 
+* }
```
Comment (beanpole135) [inline static/includes/TrueCommand/2.0/TrueCommandLDAP.md.part:46]:
```suggestion
* Domain: "dc=mycorp,dc=com"
* User ID Field: "uid" 
```

File: static/includes/TrueCommand/2.0/TrueCommandLDAP.md.part [hunk 3]
```diff
@@ -15,6 +15,64 @@ Multiple LDAP servers and Domains can be added.
...
+Once logged in, a search is then performed to find the domain of the user currently requesting to login, and then a second bind is attempted utilizing the user domain and provided password.
+
+Example:
+
+* { 
+* "domain" : "dc=mycorp,dc=com", 
+* "user_id_field" : "uid", 
+* "bind_user_domain" : "uid=binduser,cn=read-only-bind,dc=mycorp,dc=com", 
+* "bind_user_password" : "pre-shared-key", 
+* }
```
Comment (beanpole135) [inline static/includes/TrueCommand/2.0/TrueCommandLDAP.md.part:62]:
```suggestion
* Domain: "dc=mycorp,dc=com"
* User ID Field: "uid"
* BIND User Domain: "uid=binduser,cn=read-only-bind,dc=mycorp,dc=com"
* BIND Password: "pre-shared-key"
```

File: static/includes/TrueCommand/2.0/TrueCommandLDAP.md.part [hunk 4]
```diff
@@ -15,6 +15,64 @@ Multiple LDAP servers and Domains can be added.
...
+
+When "bobby.singer" attempts to login, TrueCommand will establish an SSL-secure connection to the LDAP server, and then the "bind_user_domain" and "bind_user_password" settings are used to perform an initial bind utilizing pre-known settings from your LDAP provider. Once bound, a search is then performed for the user matching "uid=bobby-singer", but only within the subdomains that include the "domain" setting ("dc=mycorp,dc=com" in this example). If a user is found, the full user domain string from the search result is then used to initialize a second bind along with the user-provided password. If successful, then the user authentication has been verified and Bobby Singer is allowed access to TrueCommand.
+
+
+### SSL/TLS Connection Info
+
+If the LDAP server uses an SSL certificate generated by a custom certificate authority (CA), then one of two things must occur before the LDAP server can be used by TrueCommand:
+
+* (Option 1) The custom certificate authority must be registered with TrueCommand via the {{< api-link "ssl/ca_import" >}} API call.
```
Comment (beanpole135) [inline static/includes/TrueCommand/2.0/TrueCommandLDAP.md.part:71]:
```suggestion
* (Option 1) The custom certificate authority must be registered with TrueCommand via the Certificates Tab of the Administrator Settings.
```

**Review by beanpole135 — CHANGES_REQUESTED** [review]
Cleanup some of the formatting.
Since you replaced the API-specific terminology with the UI terminology, you need to ensure you do that same conversion everywhere else too. I fixed a couple of them, but there are some more places which reference API fields.
The table also looks a bit funny with the example as a second table row instead of within the same table cell. You should be able to multiple multiple lines of text within a single table cell by wrapping the contents within an HTML element like `<span>text..<br>on line 2</span>`.

**Comment by BonnieF** [comment]
I asked @beanpole135  to check for accuracy. I need to sync with @Mrt134  about an edit choice.

File: static/includes/TrueCommand/2.0/TrueCommandLDAP.md.part [hunk 5]
```diff
@@ -15,6 +15,54 @@ Multiple LDAP servers and Domains can be added.
...
+| User ID Field (string) | Domain fieldname to use for user-matching. The default value is *User ID FIELD*. Another field commonly-used is "cn" (common name) |
+| Group ID Field (string) | The domain fieldname to use when searching for a group name. The default value is "cn" (common name). |
+| BIND User Domain (string) | <span>The full domain setting for a pre-authenticated bind to the server.<br>Example: "uid=binduser,cn=read-only-bind,dc=mycorp,dc=com"<br>For an unauthenticated bind set this field to just a name (example: "truecommand-bind"). This is sometimes used for logging purposes on the LDAP, but otherwise is not validated.</span> |
+| Bind Password (string) | The password to use for the bind user For an unauthenticated bind, leave this field blank while setting the *BIND User Domain* to a non-empty value. |
+
+### LDAP connection options
+
+There are two common methods of validating an LDAP user which TrueCommand supports:
```
Comment (beanpole135) [inline static/includes/TrueCommand/2.0/TrueCommandLDAP.md.part:31]:
```suggestion
TrueCommand supports two common methods of validating LDAP user credentials:
```

File: static/includes/TrueCommand/2.0/TrueCommandLDAP.md.part [hunk 6]
```diff
@@ -15,6 +15,54 @@ Multiple LDAP servers and Domains can be added.
...
+
+*Direct Bind*
```
Comment (beanpole135) [inline static/includes/TrueCommand/2.0/TrueCommandLDAP.md.part:33]:
```suggestion
#### Direct Bind
```

File: static/includes/TrueCommand/2.0/TrueCommandLDAP.md.part [hunk 7]
```diff
@@ -15,6 +15,54 @@ Multiple LDAP servers and Domains can be added.
...
+In this method, the *Domain* and *User ID Field* are used to create a static domain string which is then utilized to attempt to authenticate the user.
+
+Example:
+
+* Domain: "dc=mycorp,dc=com"
+* User ID Field: "uid" 
+
+When user "bobby.singer" attempts to login, TrueCommand will establish an SSL-secure connection to the LDAP server and then attempt to bind with the static domain "uid=bobby.singer,dc=mycorp,dc=com" and the user-provided password. If successful, then the user authentication has been verified and Bobby Singer is allowed access to TrueCommand.
+
+*Indirect Bind*
```
Comment (beanpole135) [inline static/includes/TrueCommand/2.0/TrueCommandLDAP.md.part:43]:
```suggestion
#### Indirect Bind
```

File: static/includes/TrueCommand/2.0/TrueCommandLDAP.md.part [hunk 8]
```diff
@@ -15,6 +15,54 @@ Multiple LDAP servers and Domains can be added.
...
+* Domain: "dc=mycorp,dc=com"
+* User ID Field: "uid"
+* BIND User Domain: "uid=binduser,cn=read-only-bind,dc=mycorp,dc=com"
+* BIND Password: "pre-shared-key"
+
+When "bobby.singer" attempts to login, TrueCommand will establish an SSL-secure connection to the LDAP server, and then the *BIND User Domain* and *BIND Password* settings are used to perform an initial bind utilizing pre-known settings from your LDAP provider. Once bound, a search is then performed for the user matching "uid=bobby-singer", but only within the subdomains that include the "domain" setting ("dc=mycorp,dc=com" in this example). If a user is found, the full user domain string from the search result is then used to initialize a second bind along with the user-provided password. If successful, then the user authentication has been verified and Bobby Singer is allowed access to TrueCommand.
+
+
+### SSL/TLS Connection Info
+
```
Comment (beanpole135) [inline static/includes/TrueCommand/2.0/TrueCommandLDAP.md.part:59]: Put a little warning box here stating that SSL connections are *required* for AD/LDAP authentication.

**Review by beanpole135 — CHANGES_REQUESTED** [review]

File: static/includes/TrueCommand/2.0/TrueCommandLDAP.md.part (hunk 8 above)
Comment (BonnieF) [reply static/includes/TrueCommand/2.0/TrueCommandLDAP.md.part:59]: I have added the warning

**Review by beanpole135 — APPROVED** [review]

**Review by tonyriv3 — APPROVED** [review]

**Pull request merged**